manifests/server.pp

```diff
@@ -21,6 +21,8 @@
   $service_provider        = $mysql::params::server_service_provider,
   $create_root_user        = $mysql::params::create_root_user,
   $create_root_my_cnf      = $mysql::params::create_root_my_cnf,
+  $create_root_login_file  = $mysql::params::create_root_login_file,
+  $login_file              = $mysql::params::login_file,
```

Comment:
You need to create the login file first with mysql_config_editor, this creates an encrypted file.
And because the tool uses interactive prompt for the password, it's not possible to create during a puppet run.
So that's why you need to source this instead of using content.

https://dev.mysql.com/doc/refman/5.6/en/mysql-config-editor.html